setup.py

```diff
@@ -0,0 +1,134 @@
+#
+#   Copyright 2013-2022 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+import inspect
+import os
+import subprocess
+import shlex
+
+from setuptools import (
+    setup,
+    Extension,
+    find_packages,
+)
+
+import setuptools.command.build_ext
+
+
+class build_ext(setuptools.command.build_ext.build_ext):
+    """
+    Custom setuptools command class to puppet CMake.
+    """
+
+    def build_extension(self, _ext: Extension):
+        """
+        Hook called by setuptools to build the given Extension.
+
+        @param _ext: Extension to build (we only have one, so ignored).
+        """
+        self.__cmake(
+            [
+                "-S",
+                ".",
+                "-B",
+                self.build_temp,
+                "-G",
+                "Ninja",
+                # Place output artifacts where setuptools expects.
+                "--install-prefix",
+                os.path.abspath(self.build_lib),
+                "--preset",
+                "setuptools",
+            ]
+        )
+
+        self.__cmake(["--build", self.build_temp, "--target", "openassetio-python-module"])
+
+        self.__cmake(
+            [
+                "--install",
+                self.build_temp,
+                "--component",
+                "openassetio-python-module",
+            ]
+        )
+
+    def __cmake(self, args: list[str]):
+        """
+        Execute `cmake` as a subprocess with given arguments.
+
+        @param args: Command-line arguments to pass to `cmake`.
+        """
+        args = ["cmake"] + args
+        self.announce(" ".join(map(shlex.quote, args)), level=2)
+        subprocess.check_call(args, env=os.environ.copy(), stderr=subprocess.STDOUT)
+
+
+setup(
+    name="openassetio",
+    version="1.0.0a4",
```

Comment:
[question] This could be specified automatically if we use [setuptools_scm](https://github.com/pypa/setuptools_scm), which also handles bundling the repo for `sdist` packages. 

We may need to conform our git tag naming convention to [PEP 440](https://peps.python.org/pep-0440/#pre-releases) for that to work properly, though. And if we want to explicitly exclude things from the `sdist` we'd need a `MANIFEST.in` regardless, which would remove some of the appeal.

On balance I think the forcing Python package version to follow the git tag is likely to cause some pain for not much gain. But worth throwing the question out there.

Reply:
I like my sdists to be leaner than the repository itself, meaning only the files required for building the project + tests. But that's going to be up to the OpenAssetIO TSC to decide if you want to use `setuptools_scm` or not.

Reply:
> We may need to conform our git tag naming convention to [PEP 440](https://peps.python.org/pep-0440/#pre-releases) for that to work properly, though. 

We'd originally followed the examples on [semver.org](https://semver.org/#spec-item-9) for the alpha pre-releases, but it looks like `a` is just as valid as `alpha`, so no harm switching moving forwards if it makes life more consistent.

Reply:
We decided to add a follow-up ticket to deal with `sdist`s, one way or another: https://github.com/OpenAssetIO/OpenAssetIO/issues/666